specs/APIReview_AllowDrop.md

```diff
@@ -0,0 +1,103 @@
+# Background
+Currently the drag&drop functionality is default enabled in webview2 and there is no way to disable it. Some developers may want to disbale this functionality in their applications 
+based upon their scenarios. According to such feature requirements, we add the new API to provide developers with the capability to configure the drag&drop functionality.
+
+# Description
+We add a new `AllowDrop` property in `CoreWebView2Controller`. 
```

Comment:
Great point! Yes AllowExternalDrop would make a better name if this doesn't apply to intra-WebView drops.

Please also update the documentation for AllowDrop to say which of the cases Raymond mentions are covered or not by AllowDrop. Thanks!

Reply:
I'm refactoring the doc to change to AllowExternalDrop. This property only has impact on dragging and dropping something from outside webview to webview. Drag&Drop data from webview to external(outside webview) will not be impacted. For the comment above, left comments below


> Does this disable all drops (even intra-page drag/drop), or only drops from outside the WebView, or only drops from non-host processes? 
**_Fei: This only disable any drops from outside webview_**
> 
> In other words, if I set `AllowDrop` to `false`, which of these actions are blocked?
> 
> * Dragging: Which of these activities raise the `dragstart` event? 
**_Fei: dragging anything from webview to any other external things is not blocked, so all below ones are not impacted by the toggle of this AllowDrop API_**
>   
>   * Dragging selected text from WebView to Word.
>   * Dragging selected text from WebView to host app.
>   * Dragging selected text from WebView to another WebView hosted by the same app.
>   * Dragging selected text from WebView to another WebView hosted by the same app that uses the same user data folder.
>   * Dragging selected text from WebView to another frame in the same WebView.
>   * Dragging selected text from WebView to the same frame of the WebView.
> * Dropping: Which of these activities raise the `dragover` / `drop` events?
>   
>   * Drag an object from Word to the WebView.
**_Fei: Not impacted by this API since drag&drop anything from word to webview is default not supported(not supported in Edge as well)._**
>   * Drag an object from the host app into the WebView.
**_Fei: Not impacted by this API since it's default not supported._**
>   * Drag an object from one WebView into another WebView hosted by the same app.
**_Fei:  Impacted by this API, the WebView that accepts the drop should be configured AllowExternalDrop to be true._**
>   * Drag an object from one WebView to another WebView hosted by the same app that uses the same user data folder.
**_Fei: Similar as above_**
>   * Drag an object that originated in the WebView to another frame in the same WebView.
**_Fei: Not impacted by this API since it's dragging files from WebView to external_**
>   * Drag an object that originated in the WebView to the same frame of the WebView.
**_Fei: Not impacted by this API since it's dragging files from WebView to external_**
> 
> If drag/drop within the WebView is still permitted, then maybe AllowExternalDrop would be a clearer name?

Reply:
I updated the doc to list some behaviors that will be impacted by the new API.

Reply:
> Drag an object from Word to the WebView.
> Fei: Not impacted by this API since drag&drop anything from word to webview is default not supported(not supported in Edge as well).

I didn't mean Word specifically, but just using Word as an example of a separate app unrelated to the WebView host app.

I can drag a file from Explorer to Edge and it navigates to the file. Is that blocked by AllowExternalDrop?